### PR TITLE
feat: implement filter on next action

### PR DESCRIPTION
**Pull request by IlliaDerevianko**
- Add nextAction field to the prisma schema with migration
- Add column filtering for the table
- Add dropdown to filter table entries by 'Next Action'
- Change search params in URL when filtering of next action occurs
- Apply search params in URL to filter by next action

Screenshot 2024-09-03 at 17 24 11
*Filter applicant by next action "INFORM_CANDIDATE"*

### CODE REVIEW

**Review by Gum-Joe — APPROVED** [review]
:) (I didn’t look at any of the code)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -46,6 +46,8 @@ jobs:
         with:
           path: node_modules
           key: ${{ runner.os }}-nextjs-${{ hashFiles('package-lock.json') }}
+      - name: Prisma generate
+        run: npx prisma generate
```
Comment (procaconsul) [inline .github/workflows/ci.yml:50]: Was the type checking failing without this?

File: app/layout.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ export default function RootLayout({
 }>) {
   return (
     <html lang="en">
-      <body className={inter.className}>
+      <body className={inter.className + ' p-6'}>
```
Comment (procaconsul) [inline app/layout.tsx:21]: Shall we use [classnames](https://www.npmjs.com/package/classnames)?

File: components/TanstackTable/index.tsx
Comment (procaconsul) [inline components/TanstackTable/index.tsx:1]: This is OK modulo the remarks we made earlier about generalising this later on

File: prisma/schema/application.prisma [hunk 3]
```diff
@@ -22,6 +32,7 @@ model Application {
   tmuaPaper2Score       Decimal         @db.Decimal(2, 1)
   tmuaOverallScore      Decimal         @db.Decimal(2, 1)
   qualifications        Qualification[]
+  nextAction    NextAction        @default(TMUA_CANDIDATE)
```
Comment (procaconsul) [inline prisma/schema/application.prisma:35]: (silly one but) how about formatting this to align with the rest of the model? The autoformatter in WebStorm does the job for me.

**Review by procaconsul — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (matalex412) [reply .github/workflows/ci.yml:50]: Yes, the type check was failing before

File: app/layout.tsx (hunk 2 above)
Comment (matalex412) [reply app/layout.tsx]: Resolved in [7a0879d](https://github.com/imperial/ug-admissions/pull/3/commits/7a0879d032fd72411c6f4a8358a4bd6cae18dc33)

File: prisma/schema/application.prisma (hunk 3 above)
Comment (matalex412) [reply prisma/schema/application.prisma]: Resolved in [27caae9](https://github.com/imperial/ug-admissions/pull/3/commits/27caae9baf292fe5b258d24698129ed62193c569)

**Review by procaconsul — APPROVED** [review]

**Pull request merged**